### PR TITLE
Create a backend API to allow students apply for a position.

### PR DESCRIPTION
**Pull request by Mari-Hayashi**

### CODE REVIEW

File: src/server/src/entities/studentToJob.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Entity, Column, PrimaryGeneratedColumn, ManyToOne } from 'typeorm';
+import { Student } from '../entities/student';
+import { Job } from '../entities/job';
+
+@Entity()
+export class StudentToJob {
```
Comment (johan1505) [inline src/server/src/entities/studentToJob.ts]: I think a better name for this entity would be JobApplication

File: src/server/src/modules/student.ts [hunk 2]
```diff
@@ -95,3 +96,36 @@ export const updateStudent = async (
...
+    const studentToJobRepository = getRepository(StudentToJob);
+
+    const student = await studentRepository.findOne(studentId);
+    if (!student) throw new Error("Student does not exist.");
+
+    // Check if job exists.
+    const job = await jobRepository.findOne(jobId);
+    if (job === undefined) throw new Error("Requested job does not exist.");
+
+    // Check if student already applied for a job.
```
Comment (johan1505) [inline src/server/src/modules/student.ts]: applied for the* job

File: src/server/src/routes/student.ts [hunk 3]
```diff
@@ -64,6 +66,44 @@ router.post('/update-profile', async (req: studentRequest, res: Response) => {
...
+    passport.authenticate('jwt', { session: false }),
+    async (req: Request, res: Response) => {
+        //checks that caller is a student.
+        const { role, studentId } = req.user as JWTStudent;
+        if (role !== 'student') {
+            return res
+                .status(UNAUTHORIZED)
+                .json({ error: 'User is not a student' });
+        }
+        const { jobId } = req.body;
```
Comment (johan1505) [inline src/server/src/routes/student.ts:89]: I think you are forgetting to declare the type of jobId

**Review by johan1505 — CHANGES_REQUESTED** [review]
Left comments

File: src/server/src/routes/student.ts (hunk 3 above)
Comment (Mari-Hayashi) [reply src/server/src/routes/student.ts:89]: How do we declare the type of jobId?

Comment (johan1505) [reply src/server/src/routes/student.ts:89]: 1. Make an interface:

interface jobApplicationRequest extends Request {
    body: {
        jobId: number;
    };
}

2. Use the new interface in the route:

router.post ..... (req: jobApplicationRequest  ...

File: src/server/src/entities/studentToJob.ts (hunk 1 above)
Comment (Mari-Hayashi) [reply src/server/src/entities/studentToJob.ts]: Done.

File: src/server/src/modules/student.ts (hunk 2 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/student.ts]: Done.

File: src/server/src/routes/student.ts (hunk 3 above)
Comment (Mari-Hayashi) [reply src/server/src/routes/student.ts:89]: Done.

**Review by johan1505 — APPROVED** [review]

**Pull request merged**